File: app/src/main/java/com/example/moviesearcher/ui/details/cast/CastViewModel.java
package com.example.moviesearcher.ui.details.cast;

import android.app.Activity;
import android.os.Bundle;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.example.moviesearcher.model.data.Person;
import com.example.moviesearcher.model.services.MovieDbApiService;
import com.example.moviesearcher.util.BundleUtil;

import java.util.List;

public class CastViewModel extends ViewModel {

    private MutableLiveData<List<Person>> cast = new MutableLiveData<>();
    private MutableLiveData<List<Person>> crew = new MutableLiveData<>();
    private MutableLiveData<Boolean> loading = new MutableLiveData<>();

    public void fetch(Activity activity, Bundle args) {
        loading.setValue(true);
        new Thread(() -> {
            if (args != null) {
                new MovieDbApiService().getPeople(args.getInt(BundleUtil.KEY_MOVIE_ID),
                        (castList, crewList) -> activity.runOnUiThread(() -> {
                            cast.setValue(castList);
                            crew.setValue(crewList);
                            loading.setValue(false);
                        }));
            }
        }).start();
    }

    LiveData<List<Person>> getCast() {
        return cast;
    }

    LiveData<List<Person>> getCrew() {
        return crew;
    }

    LiveData<Boolean> getLoading() {
        return loading;
    }
}

File: app/src/main/java/com/example/moviesearcher/ui/details/cast/CastViewModel.kt
package com.example.moviesearcher.ui.details.cast

import android.app.Activity
import android.os.Bundle
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.example.moviesearcher.model.data.Person
import com.example.moviesearcher.model.services.MovieDbApiService
import com.example.moviesearcher.model.services.responses.PersonListAsyncResponse
import com.example.moviesearcher.util.KEY_MOVIE_ID

class CastViewModel : ViewModel() {

    private val _cast = MutableLiveData<List<Person>>()
    private val _crew = MutableLiveData<List<Person>>()
    private val _loading = MutableLiveData<Boolean>()

    val cast: LiveData<List<Person>> = _cast
    val crew: LiveData<List<Person>> = _crew
    val loading: LiveData<Boolean> = _loading

    fun fetch(activity: Activity, args: Bundle?) {
        _loading.value = true
        Thread(Runnable {
            if (args != null) {
                MovieDbApiService().getPeople(args.getInt(KEY_MOVIE_ID),
                        PersonListAsyncResponse { castList, crewList ->
                            run {
                                activity.runOnUiThread {
                                    _cast.value = castList
                                    _crew.value = crewList
                                    _loading.setValue(false)
                                }
                            }
                        }
                )
            }
        }).start()
    }
}